### PR TITLE
support oracle number(p,s) type

### PR DESCRIPTION
**Pull request by BingAo-Office**

### CODE REVIEW

**Review by rasifr — CHANGES_REQUESTED** [review]
I think there are some issue still. After applying the patch, I have observed the following.

1. see below query, It should be able to execute without issues.

> postgres=# select 1/2::numeric;
ERROR:  operator is only a shell: integer / numeric
LINE 1: select 1/2::numeric;
                ^
postgres=# 

2. even after setting the compatibility mode, the error is still there. 

> postgres=# set compatible_mode TO oracle ;
SET
postgres=# select 1/2::numeric;
ERROR:  operator is only a shell: integer / numeric
LINE 1: select 1/2::numeric;
                ^
postgres=# 

3. and replacing numeric with number results in ambiguous operator selection.

> postgres=# select 1/2::number;
ERROR:  operator is not unique: integer / "number"
LINE 1: select 1/2::number;
                ^
HINT:  Could not choose a best candidate operator. You might need to add explicit type casts.
postgres=#

**Comment by BingAo-Office** [comment]
Thanks for your review ,  I found many errors and redundant operators through the inspection, which led to the above patch errors, which have been repaired now

File: contrib/orafce/orafce--3.18.sql [hunk 1]
```diff
@@ -7638,284 +7333,3 @@ RETURNS numeric
...
-minvfunc = oracle.number_accum_inv,
-MFINALFUNC = oracle.number_stddev_samp
-);
-
-/* to_char */
-CREATE OR REPLACE FUNCTION oracle.to_char(oracle.number, text)
-RETURNS text
-AS $$SELECT pg_catalog.to_char($1::numeric, $2)$$
-LANGUAGE sql STABLE PARALLEL SAFE STRICT;
-/* oracle.number type support */
```
Comment (BingAo-Office) [inline contrib/orafce/orafce--3.18.sql]: All of these can be implicitly converted to numeric implementations

File: contrib/orafce/orafce--3.18.sql [hunk 2]
```diff
@@ -7254,255 +7188,14 @@ LANGUAGE internal
...
-CREATE OPERATOR oracle.^ (
-  LEFTARG   = oracle.number,
-  RIGHTARG  = oracle.number,
-  PROCEDURE = oracle.number_power,
-  COMMUTATOR = operator(oracle.^)
-);
-
---end
------------------------------------------------------------------
--- end
```
Comment (BingAo-Office) [inline contrib/orafce/orafce--3.18.sql]: All of these can be implicitly converted to numeric implementations

File: contrib/orafce/orafce--3.18.sql [hunk 3]
```diff
@@ -5736,27 +5736,6 @@ ALTER FUNCTION dbms_assert.enquote_name ( character varying ) STRICT;
...
-RETURNS numeric
-AS $$SELECT pg_catalog.round($1::numeric, $2)$$
-LANGUAGE sql IMMUTABLE STRICT;
-
-CREATE OR REPLACE FUNCTION oracle.trunc(float4, int)
-RETURNS numeric
-AS $$ SELECT pg_catalog.trunc($1::numeric, $2); $$
-LANGUAGE sql IMMUTABLE STRICT;
-
-
```
Comment (BingAo-Office) [inline contrib/orafce/orafce--3.18.sql:5759]: Adjust the position

**Comment by rasifr** [comment]
@BingAo-Office Thanks for the update but it needs to be rebased again.

**Review by rasifr — APPROVED** [review]
everything seems to be working

**Pull request merged**